### PR TITLE
Add #[doc(search_hidden)] attribute

### LINKED ISSUE #67368: [rustdoc] Add a new doc attribute that hides items from search but allow leave it clickable under publiced module
In #67346, I wanted to hide HashMap from search under `hash_map` module.
[](https://imgur.com/TkkiEV6.png)

It could be nice to have a way to 
* hide items from search but leave it under other publicized module
* display only one in search for items that is re-publiced in multiplle modules.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #67368.

r? @jyn514

### CODE REVIEW

**Comment by jyn514** [comment]
@GuillaumeGomez if you want this to be stabilized at some point it needs a tracking issue. You made https://github.com/rust-lang/rust/issues/67368 the tracking issue in the code but you also said 'fixes' in the commit. Either it needs a new issue or you need to not close the existing one.

File: src/test/rustdoc-js/search_hidden.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#![feature(doc_search_hidden)]
+
+#[doc(search_hidden)]
+/// Foo
+pub struct Bar;
```
Comment (jyn514) [inline src/test/rustdoc-js/search_hidden.rs:5]: Can you also add a test that re-exports are still shown?
```suggestion
#[doc(search_hidden)]
/// Foo
pub struct Bar;

mod inner {
  pub use Bar;
}
```
and then update `search_hidden.js`.

**Comment by GuillaumeGomez** [comment]
Ah indeed, force of habit... I'll update my first comment. Good catch, thanks!

File: src/test/rustdoc-js/search_hidden.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/test/rustdoc-js/search_hidden.rs:5]: Sure!

**Comment by jyn514** [comment]
I think this feature doesn't make sense in isolation, I'd rather have doc(canonical): https://github.com/rust-lang/rfcs/issues/3011. I'll try to write that up into a full RFC some time this week.

**Pull request closed without merging**